strapi/strapi
# test(database): init unit tests for createMetadata

### What does it do?

Adds unit tests for createMetadata

Note that it is mostly a monolithic test that takes a dump of models from a Strapi project and compares the output metadata to an expected set of metadata. This was the quickest and safest way to get 100 percent coverage of all the lines and branches that include database identifier names. In future revisions, we need to aim for breaking this up into individual and specific tests, for example for each type of relation. However, this monolithic test can still be kept as a holistic test that it works with large complex data.

### Why is it needed?

We are about to do a large refactoring of the way identifiers for the database are generated. We need to be 100% certain that the refactoring does not change any identifier in any way that we do not explicitly intend to change.

And in general, tests were lacking, so at least now we will be able to prevent any accidental breaking changes in the db metadata.

### How to test it?

Run the tests with --coverage and ensure that every part of metadata/index.ts and metadata/relations.ts that concatenate a string to form a column, index, or table name are covered.

### Related issue(s)/PR(s)

Similar PR for changes outside of `@strapi/database`: https://github.com/strapi/strapi/pull/19460
